modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -91,6 +98,69 @@ public class UserLocalServiceTest {
 	public static final AggregateTestRule aggregateTestRule =
 		new LiferayIntegrationTestRule();
 
+	@Test(expected = PasswordExpiredException.class)
+	public void testAuthenticateByEmailAddressPasswordExpired()
+		throws Exception {
+
+		String password = "password";
+
+		User user = _addRandomUserWithPassword(RandomTestUtil.randomString());
+
+		_userLocalService.updatePassword(
+			user.getUserId(), password, password, false, true);
+
+		user = _userLocalService.getUser(user.getUserId());
+
+		PasswordPolicy passwordPolicy = user.getPasswordPolicy();
+
+		passwordPolicy.setExpireable(true);
+		passwordPolicy.setMaxAge(0);
+
+		PasswordPolicyLocalServiceUtil.updatePasswordPolicy(passwordPolicy);
+
+		_userLocalService.authenticateByEmailAddress(
+			user.getCompanyId(), user.getEmailAddress(), password, null, null,
+			null);
+	}
+
+	@Test
+	public void testAuthenticateByEmailAddressWrongPasswordWithPasswordPolicy()
+		throws Exception {
+
+		String newPassword = RandomTestUtil.randomString();
+
+		String password = RandomTestUtil.randomString();
+
+		User user = _addRandomUserWithPassword(password);
+
+		PasswordPolicy passwordPolicy = user.getPasswordPolicy();
+
+		passwordPolicy.setExpireable(true);
+		passwordPolicy.setMaxAge(0);
+
+		PasswordPolicyLocalServiceUtil.updatePasswordPolicy(passwordPolicy);
+
+		int failedLoginAttempts = user.getFailedLoginAttempts();
+
+		Assert.assertEquals(
+			Authenticator.FAILURE,
+			_userLocalService.authenticateByEmailAddress(
+				user.getCompanyId(), user.getEmailAddress(), newPassword, null,
+				null, null));
+
+		try {
+			_userLocalService.authenticateByEmailAddress(
+				user.getCompanyId(), user.getEmailAddress(), password, null,
+				null, null);
+		}
+		catch (PortalException portalException) {
+			user = _userLocalService.getUser(user.getUserId());
+
+			Assert.assertEquals(
+				failedLoginAttempts + 1, user.getFailedLoginAttempts());
+		}
```

Comment:
I think no need to call _userLocalService.authenticateByEmailAddress() again. Just using 
```
_userLocalService.fetchUser(user.getUserId());
Assert.assertEquals(1, newUser.getFailedLoginAttempts());
```